refinery/data_set_manager/search_indexes.py

```diff
@@ -66,13 +66,17 @@ def prepare(self, object):
             logger.warn(e)
 
         if object.assay is not None:
-            suffix += "_" + str(object.assay.id)
+            id_suffix += "_" + str(object.assay.id)
 
-        suffix = "_" + suffix + "_s"
+        id_suffix = "_" + id_suffix + "_s"
         generic_suffix = "_generic_s"
 
         data['filename_Characteristics' + generic_suffix] = \
             re.sub(r'.*/', '', data['name'])
+        data['technology_Characteristics' + generic_suffix] = \
+            set() if (object.assay is None or
+                      object.assay.technology is None) \
+            else {object.assay.technology}
```

Comment:
Could this be cleaned up?

Its not immediately apparent why we need an empty `set()`

maybe something like:
```
technology_key = 'technology_Characteristics' + generic_suffix
if (object.assay is None orobject.assay.technology is None):
    # We need an empty set here because: ...
    data[technology_key] = set()
else:
    data[technology_key] = object.assay.technology
```
and an additional test to assert the empty set if there is no technology or Assay